docs/source/user-guide/tabular/supported_models.rst

```diff
@@ -1,107 +1,111 @@
+The prediction can be provided **either** within a model object or as pre-computed static predictions.
 
-.. Note::
-    In order to be compatible with deepchecks, the model object should adhere to one requirement:
-
-    - It has to have a a method enabling **prediction**
-
-More specifically, the model should have a ``predict`` method for regression tasks, 
-and for classification tasks also a ``predict_proba`` method, 
-which should be implemented using the scikit-learn api conventions. 
-Some checks may attempt using additional model methods if those exist, 
-for more information see :ref:`Optional Model Interface <supported_models__optional_model_interface>`.
+Whether provided from a model interface or as a pre-computed static predictions,
+the predictions must be in the following format based on the task type:
 
-Note that built-in scikit-learn classifiers and regressors, 
-along with many additional popular models types (e.g. XGBoost, LightGBM, CatBoost etc.) implement these methods 
-and are thus supported.
+* Classification: ``proba``, an `ndarray` of shape ``(n_samples, n_classes)`` containing the predicted
+  probability of each class per sample in the dataset.
+* Regression: ``pred``, an `ndarray` of shape ``(n_samples,)``, containing the predicted value
+  for each sample in the dataset.
 
+For classification tasks, it is optional to also supply ``pred`` predictions. If not supplied, Deepchecks will
+assume the predicted value is the class with the highest probability.
 
-Required Model Interface
-==========================
+Using a Model Interface
+***********************
 
-Regression
-----------
+Supported model objects should implement at least one of the following interfaces based on the task type:
 
-All that is necessary for a regression model is the ``predict`` function.
-The predict function should expect an |array-like|  of shape ``(n_samples, n_features)``
-and is expected to return an `ndarray` of shape ``(n_samples,)``, a vector containing the predicted value for each sample.
+* Classification: ``predict_proba`` method which receives an |array-like|  of shape ``(n_samples, n_features)``
+  and is expected to return an `ndarray` as specified above.
+* Regression: ``predict`` method which receives an |array-like|  of shape ``(n_samples, n_features)``
+  and is expected to return an `ndarray` as specified above.
 
-Example of a simple regression model:
-
->>> class simple_regression_model:
-...     def predict(X: pd.DataFrame) -> pd.Series:
-...         ...
+.. Note::
+    The methods are implemented using the scikit-learn api conventions therefore built-in scikit-learn
+    classifiers and regressors,
+    along with many additional popular models types (e.g. XGBoost, LightGBM, CatBoost etc.) are supported.
 
-Instead of passing a model object you can pass the precomputed predictions as keyword arguments to the run function as shown here:
+Optional Model Interface
```

Comment:
These are less interesting than the code examples for "wrappers", would put them in a less major place